controllers/storagecluster/cephcluster.go

```diff
@@ -123,6 +123,9 @@ func (obj *ocsCephCluster) ensureCreated(r *StorageClusterReconciler, sc *ocsv1.
 		if err := verifyMonitoringEndpoints(monitoringIP, monitoringPort, r.Log); err != nil {
 			r.Log.Error(err, "Could not connect to the Monitoring Endpoints.",
 				"CephCluster", klog.KRef(sc.Namespace, sc.Name))
+			// reset the external resources cache to nil,
+			// when monitoring endpoints are not reachable
+			externalOCSResources[sc.UID] = nil
```

Comment:
The short answer: We do not. 
The long answer:  
1. We return an error, which in turn will queue another event,
2. A new reconcile for the new event is triggered
3. The code in `ensure_created` on `external _resources.go` sees that we have no cache line for this storageclsuter and will fetch again

Reply:
My question revolves around the `3` rd step.
When we reset a map variable's (particular) key value to nil, we are not resetting any cache (because the map var `externalOCSResources` will always start as an empty map for each reconcile). [this is my understanding]

In order for the `3` rd step (above mentioned) to be true, the assumption is it has to enter `ensure_created` (on `external_resources.go` file) again in the **same** reconcile call (after we reset map of that storagecluster-id to `nil`)

Reply:
Again thanks @nb-ohad . Now I got the context that the map var which we edit is a global/package level one, which won't be reset on each reconcile. Then this solves the confusion.